src/Connect/Payments/CreditCard.php

```diff
@@ -287,4 +287,129 @@ public static function orderPayScript($template_name)
+
+        $generate_transient = get_transient('product_installment_enabled');
+
+        if ($cc_enabled_installments === 'yes' && !$generate_transient) {
+            $product_ids = wc_get_products(array(
+                'status' => 'publish',
+                'limit' => -1,
+                'return' => 'ids',
+            ));
+
+            foreach ($product_ids as $product_id) {
+                self::updateProductInstallmentsTransient($product_id);
+            }
+
+            set_transient('product_installment_enabled', true);
+
+        } else if($cc_enabled_installments === 'no'){
+            $product_ids = wc_get_products(array(
+                'status' => 'publish',
+                'limit' => -1,
+                'return' => 'ids',
+            ));
+
+            foreach ($product_ids as $product_id) {
+                delete_transient('product_installment_info_' . $product_id);
+                delete_transient('wc_related_' . $product_id);
+                delete_transient('timeout_wc_related_' . $product_id);
+            }
+
+            set_transient('product_installment_enabled', false);
+        }
+    }
+
+    /**
+     * Function to update the transient when the product is updated
+     * @param int $product_id The ID of the product being updated
+     * @return void
+     */
+    public static function updateProductInstallmentsTransient($product_id) 
+    {
+        delete_transient('product_installment_info_' . $product_id);
+
+        $product = wc_get_product($product_id);
+
+        if (!$product) return;
+
+        $cc_enabled_installments = Params::getConfig('cc_enabled_installment');
+
+        if ($cc_enabled_installments === 'yes') {
+
+            $default_installments = Params::getInstallments($product->get_price(), '555566');
+
+            if ($default_installments) {
+                $installment_info = '';
+                
+                $iteration = 0;
+                foreach ($default_installments as $installment) {
+
+                    if ($iteration % 2 == 0) {
+                        $installment_info .= '<tr>';
+                    }
+
+                    $amout = number_format($installment['installment_amount'], 2, ',', '.');
+                    $total_amount = number_format($installment['total_amount'], 2, ',', '.');
+
+                    $installment_info .= '<td>' . $installment['installments'] . 'x de R$ ' . $amout . ($installment['interest_free'] ? '<small> Sem juros</small>' : '<small> Total: R$' . $total_amount . '</small>') . '</td>';
+                    
+                    if ($iteration % 2 != 0 || $iteration == count($default_installments) - 1) {
+                        $installment_info .= '</tr>';
+                    }
+                    
+                    $iteration++;
+                }
+            }
+
+            set_transient('product_installment_info_' . $product_id, $installment_info, WEEK_IN_SECONDS);
```

Comment:
Me parece que definiu que a duração do cache é de 7 dias. Peço que altere para 1 ano. Lembrando que o cache deve ser apagado se o produto sofrer alteração de preço ou se uma das configurações de parcelamento for alterada no painel.